controllers/spiaccesstokenbinding_controller.go

```diff
@@ -81,9 +85,13 @@ type SPIAccessTokenBindingReconciler struct {
 // SetupWithManager sets up the controller with the Manager.
 func (r *SPIAccessTokenBindingReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.syncer = sync.New(mgr.GetClient())
+	r.finalizers = finalizer.NewFinalizers()
+	if err := r.finalizers.Register(linkedSecretsFinalizerName, &linkedSecretsFinalizer{client: r.Client}); err != nil {
+		return fmt.Errorf("failed to register the linked secrets finalizer: %w", err)
+	}
+
 	err := ctrl.NewControllerManagedBy(mgr).
 		For(&api.SPIAccessTokenBinding{}).
-		Owns(&corev1.Secret{}).
```

Comment:
as we discussed, please check the behavior after this change. I think we want to keep watching the secrets so we reconcile them back if there is some external change to them.

Reply:
Good point. I've added the secrets watch.